lib/AST/LookupVisibleDecls.cpp

```diff
@@ -805,10 +805,10 @@ class OverrideFilteringConsumer : public VisibleDeclConsumer {
           OtherSignatureType = CT->getCanonicalType();
       }
 
-      if (conflicting(FoundSignature, OtherSignature, true) &&
-          (FoundSignatureType == OtherSignatureType ||
-           FoundSignature.Name.isCompoundName() !=
-             OtherSignature.Name.isCompoundName())) {
+      if (conflicting(VD->getDeclContext()->getASTContext(),
```

Comment:
No need to jump up to the DeclContext here; Decls also have a `getASTContext()`. (But it'd be fastest to get it from the module `M`, which always has a direct reference.)